### PR TITLE
Support featured image

### PR DESCRIPTION
**Pull request by rise-erpelding**
## Description

Utilizes the featured image on a WordPress post in the post list (archive page) as a thumbnail and in the post.

Addresses #75 

## To Validate

1. Make sure all PR Checks have passed
2. Pull down this branch
3. Run `npm start`
4. If you have posts with images, keep the database you already have, or add your own, or use the db backup attached below.
5. Navigate to localhost:8000 to check the archive page, check that thumbnails render and that styling seems appropriate.
6. Navigate to a post page that has a thumbnail image and check that image renders there and that styling seems appropriate.
7. Add a post with a featured image and confirm that image appears on archive page and on post page as expected.
8. Add alt text to a featured image if it does not have one already (you can do this in the Post by clicking on "Replace" to replace the image, then update the details and save). Confirm that the alt text is visible on the image for the archive page and for the post page.

[lorem-ipsum-with-img-2023-10-23T23-16-57_UTC.sql.gz](https://github.com/sparkbox/sparkpress-wordpress-starter/files/13090152/lorem-ipsum-with-img-2023-10-23T23-16-57_UTC.sql.gz)

### CODE REVIEW

File: src/php/views/partials/content-single.twig [hunk 1]
```diff
@@ -1,8 +1,19 @@
-<article id="post-{{ post.id }}">
-	<h2>
-		{{ post.title }}
-	</h2>
+<article id="post-{{ post.id }}" class="post-preview">
```
Comment (rise-erpelding) [inline src/php/views/partials/content-single.twig:1]: Naming things is hard, I wasn't sure about `post-preview` here, if you have a better idea I am happy to hear it!

File: src/php/views/partials/content-single.twig [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+			{% endif %}
+		/>
+	{% endif %}
 	<div>
-		{{ post.preview }}
+		<h2 class="post-preview__title">
+			{{ post.title }}
+		</h2>
+		<div>
+			{{ post.preview.chars(180) }}
```
Comment (rise-erpelding) [inline src/php/views/partials/content-single.twig:16]: Timber says that the default preview length is 50 words but in randomly spot-checking some of my posts I wasn't sure if this is accurate. I wanted to make the post previews more uniform in length so I went for setting a limit on the number of characters instead. Spot-checking again it still seems like there still are some (longer than 180 char) posts that preview less than 180 characters, but it's a little bit more consistent 🤷‍♀️

File: src/php/views/page.twig [hunk 3]
```diff
@@ -3,6 +3,12 @@
 {% block content %}
 
 	<div class="obj-width-limiter">
+		<img
+			src="{{ post.thumbnail.src|resize(1024) }}"
```
Comment (jonoliver) [inline src/php/views/page.twig:7]: Thank you for including the resize! 🏆 

It looks like `obj-width-limiter` uses `min` to ensure that the width does not exceed 60rem/960px. So I think we can update the value to `960`, unless there was a specific reason for `1024`.

Comment (jonoliver) [reply src/php/views/page.twig:7]: I think using a hardcoded resize value is perfect for this PR. There is also an opportunity to leverage the `picture`/`source` tags for responsive images, but I will create a separate issue for that.

File: src/php/views/partials/content-single.twig (hunk 1 above)
Comment (jonoliver) [reply src/php/views/partials/content-single.twig:1]: I don't see anything else conflicting with this class name, sounds good to me! 👍

File: src/php/views/partials/content-single.twig (hunk 2 above)
Comment (jonoliver) [reply src/php/views/partials/content-single.twig:16]: I generated a list of numbers to test this, removing the `.chars(180)`, and it looks like it truncated right at 50 words:
image

There are probably edge cases, but I think we can stick with the default behavior for this PR, and leave it up to the user to determine how to truncate.

**Review by jonoliver — CHANGES_REQUESTED** [review]
Nice work Rise! To capture the requested changes:

- [x] Update `resize` value to `960`
- [ ] Remove character truncation from `post.preview`

I added #111 to capture making the images responsive at different viewport widths.

**Review by jonoliver — APPROVED** [review]
Thank you Rise!
